backend/tests/services/test_task_knowledge_base_sync.py

```
@@ -238,20 +240,25 @@ def test_subtask_kb_takes_priority(self, mock_db):
-                )
-
-                # Should use only subtask KB (10), not task-level (20, 30)
-                mock_kb_tool.assert_called_once()
-                call_args = mock_kb_tool.call_args
-                assert call_args[1]["knowledge_base_ids"] == [10]
-                assert len(kb_result.extra_tools) == 1
+                with patch.object(
+                    knowledge_share_service,
+                    "is_user_restricted_observer_for_any_kb",
+                    return_value=False,
+                ) as mock_restricted:
```

Ruff (0.15.5)

[error] Local variable `mock_restricted` is assigned to but never used

Remove assignment to unused variable `mock_restricted`

(F841)